ldk/javascript/src/whisper/index.ts

```diff
@@ -265,6 +284,146 @@ export interface WhisperAptitude {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Checkbox;
+    case WhisperComponentType.Email:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Email;
+    case WhisperComponentType.Link: {
+      const { onClick } = component;
+      if (onClick) {
+        return {
+          ...component,
+          onClick: (error, whisper) => {
+            onClick(error, convertGojaWhisper(whisper));
+          },
+        } as OliveHelps.Link;
+      }
+      return component as OliveHelps.Link;
+    }
+    case WhisperComponentType.Divider:
+    case WhisperComponentType.ListPair:
+    case WhisperComponentType.Markdown:
+    case WhisperComponentType.Message:
+      return component;
+    case WhisperComponentType.Number:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.NumberInput;
+    case WhisperComponentType.Password:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Password;
+    case WhisperComponentType.RadioGroup:
+      return {
+        ...component,
+        onSelect: (error, param, whisper) => {
+          component.onSelect(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.RadioGroup;
+    case WhisperComponentType.Select:
+      return {
+        ...component,
+        onSelect: (error, param, whisper) => {
+          component.onSelect(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Select;
+    case WhisperComponentType.Telephone:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.Telephone;
+    case WhisperComponentType.TextInput:
+      return {
+        ...component,
+        onChange: (error, param, whisper) => {
+          component.onChange(error, param, convertGojaWhisper(whisper));
+        },
+      } as OliveHelps.TextInput;
+    default:
+      throw new Error('Unexpected component type');
+  }
+}
+
+function convertComponent(component: Components): OliveHelps.Components {
+  switch (component.type) {
+    case WhisperComponentType.CollapseBox:
+      return {
+        label: component.label,
+        open: component.open,
+        children: component.children.map(convertChildComponent),
+        type: WhisperComponentType.CollapseBox,
+      };
+    default:
+      return convertChildComponent(component);
+  }
+}
+
+function convertToGojaWhisper(whisper: UpdateWhisper): OliveHelps.UpdateWhisper;
```

Comment:
I like the naming convention `internal` which would signify `OliveHelps` flavor of types and `external` that would signify our LDK aptitude types.

Reply:
Sure! I'll make that change